api/receiver/create_run.go

```diff
@@ -23,12 +23,10 @@ const InternalUsername = "_processor"
 
 // HandleResultsCreate handles the POST requests for creating test runs.
 func HandleResultsCreate(a API, s checks.API, w http.ResponseWriter, r *http.Request) {
-	username, password, ok := r.BasicAuth()
-	if !ok || username != InternalUsername || !a.AuthenticateUploader(username, password) {
-		http.Error(w, "Authentication error", http.StatusUnauthorized)
+	if !AuthenticateProcessor(a, r) {
+		http.Error(w, "This is a private API.", http.StatusUnauthorized)
```

Comment:
s/private/authenticated/ and point to documentation?

Reply:
No this is indeed private in the sense that no one should ever use it except the processor service.